Assets/Scripts/Character/Hero/HeroBehavior.cs

```diff
@@ -113,26 +113,30 @@ private void Update()
 
         private void OnCollisionEnter2D(Collision2D collision)
         {
-            if (collision.gameObject.tag == Enums.Tags.Enemy.ToString() ||
-                collision.gameObject.tag == Enums.Tags.EnemyWeapon.ToString() ||
-                collision.gameObject.tag == Enums.Tags.Trap.ToString())
+            OnTriggerEnter2D(collision.collider);
```

Comment:
Don't call this method like this first off.  Second, don't just wholly delete a unity method like this.  If you need to add something that happens via a trigger then only add that logic, don't delete the logic for normal colliders.  This breaks assumptions of how the code behaves.   If you want things to behave similarly with DRY, then make a new method that has those particular parts.